component/client/core/ClientChat.tsx

```diff
@@ -11,6 +11,7 @@ interface IClientChatState {
 interface IClientChatProps {
   username: string;
   url: string;
+  switchRoom: (url: string) => void;
```

Comment:
I don't think this is necessary.
ClientChat probably does not need to switch to a different URL.
The URL is pushed into it.